src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentNodeWriter.cs

```diff
@@ -340,6 +333,40 @@ protected void WriteComponentTypeInferenceMethod(CodeRenderingContext context, C
         }
     }
 
+    protected static void WriteComponentAttributeName(CodeRenderingContext context, ComponentAttributeIntermediateNode attribute, bool allowNameof = true)
+    {
+        if (allowNameof && attribute.BoundAttribute?.ContainingType is string containingType)
+        {
+            containingType = attribute.Annotations[ComponentMetadata.Component.ConcreteContainingType] as string ?? containingType;
+
+            // nameof(containingType.PropertyName)
```

Comment:
Consider adding a comment with motivation why we are doing this so it's clear to future readers (e.g., quoting from the PR's description "This allows things like Find All References to work in the IDE as we have an actual reference to the parameter.")